### PR TITLE
Jena 1305 ES support

### PR DESCRIPTION
**Pull request by anujgandharv**
Implementation of https://issues.apache.org/jira/browse/JENA-1305

### CODE REVIEW

File: jena-text/pom.xml [hunk 1]
```diff
@@ -81,6 +81,87 @@
       <artifactId>lucene-queryparser</artifactId>
     </dependency>
 
+      <dependency>
+          <groupId>org.elasticsearch</groupId>
+          <artifactId>elasticsearch</artifactId>
+          <version>5.2.1</version>
```
Comment (osma) [inline jena-text/pom.xml]: ES version should be specified only once as a property in jena-parent/pom.xml where also e.g. the lucene version is set. All other references to the ES version should reference that property and not hardcode version 5.2.1.

File: jena-text/src/main/java/examples/JenaESTextExample.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package examples;
+
+import org.apache.jena.query.Dataset;
+import org.apache.jena.query.DatasetFactory;
+
+/**
+ * Simple example class to test the {@link org.apache.jena.query.text.assembler.TextIndexESAssembler}
+ * For this class to work properly, an elasticsearch V 5.2.1 node should be up and running, otherwise it will fail.
```
Comment (osma) [inline jena-text/src/main/java/examples/JenaESTextExample.java]: Don't mention the specific ES version in java code or comments

File: jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+     */
+    private String INDEX_NAME;
+
+    static final String CLUSTER_NAME = "cluster.name";
+
+    static final String NUM_OF_SHARDS = "number_of_shards";
+
+    static final String NUM_OF_REPLICAS = "number_of_replicas";
+
+    private static Logger LOGGER      = LoggerFactory.getLogger(TextIndexES.class) ;
```
Comment (osma) [inline jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java]: I think the convention is to use lowercase variable names / fields unless they are declared final

File: jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    @Override
+    public void addEntity(Entity entity) {
+        LOGGER.debug("Adding/Updating the entity in ES");
+
+        try {
+            XContentBuilder builder = jsonBuilder()
+                    .startObject();
+
+            if (docDef.getGraphField() != null) {
+                builder = builder.field(docDef.getGraphField(), entity.getGraph());
```
Comment (osma) [inline jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java]: If you plan to support graph fields, then the code should also be prepared to create separate entities / ES documents for each graph even if they have the same URI.

File: jena-text/src/main/java/org/apache/jena/query/text/assembler/TextIndexESAssembler.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            String replicationFactor = GraphUtils.getAsStringValue(root, pReplicas);
+            if(replicationFactor == null || replicationFactor.isEmpty()) {
+                LOGGER.warn("replicas property is not specified. Defaulting to '1'");
+                replicationFactor = "1";
+            }
+
+            String indexName = GraphUtils.getAsStringValue(root, pIndexName);
+            if(indexName == null || indexName.isEmpty()) {
+                LOGGER.warn("index Name property is not specified. Defaulting to 'test'");
+                indexName = "test";
```
Comment (osma) [inline jena-text/src/main/java/org/apache/jena/query/text/assembler/TextIndexESAssembler.java]: Suggestion: the index name could default to "jena-text"

File: jena-text/pom.xml (hunk 1 above)
Comment (anujgandharv) [inline jena-text/pom.xml]: Done

File: jena-text/src/main/java/examples/JenaESTextExample.java (hunk 2 above)
Comment (anujgandharv) [inline jena-text/src/main/java/examples/JenaESTextExample.java]: Done

File: jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java (hunk 3 above)
Comment (anujgandharv) [inline jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java]: Done

File: jena-text/src/main/java/org/apache/jena/query/text/assembler/TextIndexESAssembler.java (hunk 5 above)
Comment (anujgandharv) [inline jena-text/src/main/java/org/apache/jena/query/text/assembler/TextIndexESAssembler.java]: Done

File: jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java (hunk 4 above)
Comment (anujgandharv) [inline jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java]: OK. I can make a change to the code. Can you give me an example Sparql text query that i can use to test the changes please.

**Comment by osma** [comment]
@anujgandharv After your latest commit I see lots of changes to code that is unrelated to jena-text, e.g. ARQ and Riot parser classes. Did you commit some changes to this branch by mistake? Can you please check and revert those changes from this branch?

**Comment by osma** [comment]
Ah, hmm, those changes apparently came from the master branch, it's just GitHub showing those changes as if they would be part of this PR. Could you please rebase on top of current master?

**Comment by osma** [comment]
Please ignore my earlier comments, I was looking at the wrong PR. Sorry.

**Pull request closed without merging**